### ISSUE TITLE
Error with partial snippets

### ISSUE THREAD

**Issue by levouh** [L3MON4D3/LuaSnip#185]
```
lua/luasnip/extras/init.lua:118: attempt to call local 'fn' (a table value)
```

when using:

```
  s({ trig = "ymd", name = "Current date", dscr = "Insert the current date" }, {
    p(os.date, "%Y-%m-%d"),
  })
```

**Comment by levouh** [comment]
The most related breaking change would seemingly be [here](https://github.com/L3MON4D3/LuaSnip/issues/81#issuecomment-924127008), but I don't see that that relates to partial nodes. Also a very similar example exists in the `Examples/snippets.lua` so I would expect this to work as well.

**Comment by L3MON4D3** [comment]
Whoops, yes, you're completely right, the problem is that in
```lua
	partial = function(func, ...)
		return F(function(_, fn, ...)
			return fn(...)
		end, {}, func, ...)
	end,
```
`fn` get's the snippet and not the first user-arg.
Fixed with 39c0eca
Thank you!
